src/idiomatic/leveraging-the-type-system/raii/scope_guards.md

```diff
@@ -0,0 +1,134 @@
+# Scope Guards
+
+A scope guard makes use of the `Drop` trait to run a given closure when it goes
+out of scope.
+
+```rust,editable,compile_fail
+use scopeguard::{ScopeGuard, guard};
+use std::{
+    fs::{self, File},
+    io::Write,
+};
+
+fn conditional_success() -> bool {
+    true
+}
+
+fn main() {
+    let path = "temp.txt";
+    let mut file = File::create(path).expect("cannot create file");
+
+    // Write something to the file
+    writeln!(file, "temporary data").unwrap();
+
+    // Create a scope guard to clean up the file unless we defuse it
```

Comment:
I think this should be put right after `let mut file`, in order to:

- emphasize that `file` and `cleanup` go together,

- to ensure that `file` gets deleted even if we have a problem in `writeln!().unwrap()`.